Support multiple joysticks

SamPersson opened the pull request:
I added support for multiple joysticks, and cleaned up the code somewhat in the process.

The changes will break existing blueprints, but the interface should be similar enough that it should be easy to adapt them.

SamPersson committed:
Add support for multiple joysticks

SamPersson committed:
Separated joystick state and joystick info in the blueprint interface
Also made them into USTRUCTs to be able to send them with events. (Creating
UObjects without a parent seems to be bad.)

SamPersson committed:
Update readme.md

SamPersson committed:
Wrong axis value sent to event

getnamo commented:
I like the struct solution, makes it easy to 'break' it and see each variable attached to it. This commit is solid.

SamPersson committed:
Merge branch 'master' of https://github.com/samiljan/UEJoystickPlugin

tsky1971 committed:
complete recrating getting and processing data. all directx stuff where removed

for me this state works good enought for the things i want to do - but i will work further on this.

sorry for the breaking of interfaces etc.

tsky1971 committed:
change comment

tsky1971 committed:
add precompiled sdl2.dll

tsky1971 committed:
add thirdparty source for compiling to other versions as mine.

tsky1971 committed:
change comment

tsky1971 committed:
sorry ikarus76...

tsky1971 committed:
correct the naming of keys
you are now able to use the input mapping system - don´t now if it works if your devices change

tsky1971 committed:
add devicesdl class

tsky1971 committed:
Ahrg.... failed to copy to the right place - sorry

tsky1971 committed:
Ahrg.... failed to copy to the right place - sorry

tsky1971 committed:
wrong place

tsky1971 committed:
correct the paths

tsky1971 committed:
first steps for hotplugin

SamPersson committed:
Switch to event-based input and improve hot plugging and "controller" support

tsky1971 committed:
Merge pull request #1 from samiljan/eventbased

Switch to event-based input and improve hot plugging and "controller" support

Seems that your changes clean up the code a lot. Some changes was on my planing too. I am primary working on multiplayer support which wasn´t work for me. 
You changed it to an eventbased blocking method. I thought about that, but i don´t if the module has it own thread !? May be the "while" will block the hole handling...I think i was wrong,

tsky1971 committed:
merge the hotplug branch to the master
change the md file

SamPersson committed:
Add devices on plugin startup again

SamPersson committed:
Made FKey names more consistent with the built in FKeys, and changed hat input back to two axes

SamPersson committed:
Switched to IInputDeviceModule based plugin

SamPersson committed:
Add MapJoystickDeviceToPlayer

tsky1971 committed:
Merge pull request #2 from samiljan/sdl

Sdl

tsky1971 committed:
Merge pull request #3 from samiljan/IInputDeviceModule

IInputDeviceModule based plugin

for it looks very clean. Better than i could do.
Seems you begin the forcefeedback implementation...

tsky1971 committed:
- clean up
- addDevice return per default true otherwise we got wrong messages

bug:
- i got from my thruster sometimes value greater abs(1) ex. 1.93 etc. Perhaps the update interval is not high enought so we miss new values if we change the throttle to fast ?

SamPersson committed:
Increase width of inputsettings dropdownlists in a hacky way

SamPersson committed:
Only replace inputsettings when in editor

SamPersson committed:
Merge remote-tracking branch 'tsky/master' into IInputDeviceModule

tsky1971 committed:
- integrate samiljan´s last namesize hack
- correct/add conversion from LOCTEXT=TEXT to toString()

SamPersson committed:
4.6 compatibility for the inputsettings customization

tsky1971 committed:
Merge remote-tracking branch 'github_samiljan/IInputDeviceModule'

Conflicts:
	Plugins/JoystickPlugin/Source/JoystickPlugin/JoystickPlugin.Build.cs
	Plugins/JoystickPlugin/Source/JoystickPlugin/Private/InputSettingsCustomization.cpp

SamPersson committed:
Cleanup

SamPersson committed:
Cleanup and follow Epic code style

SamPersson committed:
More cleanup and Epic code style

SamPersson committed:
Clean up blueprint events

SamPersson committed:
Ignore GameController by default, and prepare for Linux support

SamPersson committed:
Better blueprint categories

SamPersson committed:
Add existing joysticks on load, the connected events are not triggered on mac at least

SamPersson committed:
Cleaned up build file

SamPersson committed:
Removed manual precompiled header (UE figures one out automatically)

tsky1971 committed:
Merge remote-tracking branch 'github_samiljan/IInputDeviceModule'

SamPersson committed:
Dll is needed

SamPersson committed:
Allow building for other platforms

SamPersson committed:
Link SDL statically to avoid need to copy dll to output dir

SamPersson committed:
Copy paste errors

tsky1971 committed:
Merge remote-tracking branch 'github_samiljan/IInputDeviceModule'

tsky1971 committed:
change parameter name "InTaretName" in "InProjectFilePath" to work with 4.8

tsky1971 committed:
Merge remote-tracking branch 'origin/IInputDeviceModule'

tsky1971 committed:
change to run with 4.11 and later

RulesCompiler.GetModuleFilename has changed to  RulesCompiler.GetFileNameFromType(this.GetType()

i removed the line for sdl2.dll - the InProjectFilePath does not exist anymore ?!

tsky1971 committed:
add replacement of dots in device and productname

tsky1971 committed:
wip fix name

tsky1971 committed:
compile under linux

tsky1971 committed:
add sdl2 bin distro win64

tsky1971 committed:
add modules

tsky1971 committed:
wip using sdl2 bin distro for win and source third party version on linux

tsky1971 committed:
wip using sdl2 bin distro for win and source third party version on linux

tsky1971 committed:
remove old stuff

tsky1971 committed:
add new working stuff

tsky1971 committed:
move doc

tsky1971 committed:
format style

tsky1971 committed:
add using on linux

tsky1971 committed:
add compiled linux version for testing

tsky1971 committed:
update linux docu

tsky1971 committed:
update path with /

tsky1971 committed:
remove hg temp branch

tsky1971 committed:
add patched sdl cmakefile.txt - changes for compiling with VC2015

tsky1971 committed:
some batchfiles to make life easyier

tsky1971 committed:
add precompiled with vc2015 sdl2 dll

tsky1971 committed:
change compiled plugins vc2015 and ue4 4.10

tsky1971 committed:
for windows is ... change sdl2 using to dynamic. Compiling with static in VC2015 seems to be journey of different CRT´s...

tsky1971 committed:
change sdl2 static lib compiled with vc2015

tsky1971 committed:
Update readme.md

tsky1971 committed:
compiled version for 4.10

tsky1971 committed:
update vc 2015 ue4.10

tsky1971 committed:
try to make better dokumentation

tsky1971 committed:
one step forgot

tsky1971 committed:
remove 2.0.3 version

tsky1971 committed:
- remove 2.0.3 version
- change the batch files to ask for installing cmake and/or mercurial

tsky1971 committed:
change install text

Commit:
Improving Readme markdown and flow

Commit:
Spell check

tsky1971 committed:
Merge pull request #4 from Deepfreeze32/master

Readme improvements

SamPersson committed:
Added licence

tsky1971 committed:
Merge pull request #5 from samiljan/license

Do you think a license like this would be okay? Yes, looks good.

SamPersson committed:
Merge pull request #2 from tsky1971/master

Update master from tsky

SamPersson committed:
Update readme.md
